File: LeetCode/Study Plan/Practice Problems/Easy/Ugly Number/Cpp/ugly_number.cc
#include <iostream>

bool ugly_number(int n) {
    if (n==0) {
        return false;
    }
    while(!(n%2)) n = n/2;
    while(!(n%3)) n = n/3;
    while(!(n%5)) n = n/5;
    return n == 1 ? true : false;
}

int main() {
    std::cout<<ugly_number(6)<<"\n";
}

File: LeetCode/Study Plan/Practice Problems/Easy/Ugly Number/Go/ugly_number.go
package main

import "fmt"

func ugly_number(n int) bool {
	for n%2 == 0 {
		n = n / 2
	}
	for n%3 == 0 {
		n = n / 3
	}
	for n%5 == 0 {
		n = n / 5
	}
	return n == 1
}

func main() {
	fmt.Println(ugly_number(6))
}
